### PR TITLE
Add state icon stories, documentation and tests

### PR DESCRIPTION
**Pull request by JoanaMMoreira**

### CODE REVIEW

File: packages/core/stories/state-icon.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  argTypes: {
+    size: {
+      options: ICON_NAMED_SIZES,
+      control: { type: "select" },
+      defaultValue: ICON_NAMED_SIZES[0],
+    },
+  },
+} as ComponentMeta<typeof StateIcon>;
+
+const states: ValidationState[] = ["error", "success", "warning", "info"];
```
Comment (origami-z) [inline packages/core/stories/state-icon.stories.tsx:21]: Feels useful to be exported from the source code

File: packages/core/stories/state-icon.doc.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    viewMode: "docs",
+  }}
+/>
+
+# State icon
+
+A state icon can be used on its own or within another component (e.g. `Tooltip`) to help convey a message.
+
+<Canvas>
+  <Story id="core-stateicon--default-state-icon" />
```
Comment (origami-z) [inline packages/core/stories/state-icon.doc.stories.mdx:31]: Purely from documentation perspective, could look nice if we have bigger size

File: packages/core/stories/state-icon.doc.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+# API
+
+```
+import { StateIcon } from "@jpmorganchase/uitk-core";
+```
+
+## Props
+
+<ArgsTable of={StateIcon} />
```
Comment (origami-z) [inline packages/core/stories/state-icon.doc.stories.mdx:48]: `size` has a default value from `Icon`, not sure why it's not shown on the props table default column

File: packages/core/stories/state-icon.doc.stories.mdx (hunk 2 above)
Comment (JoanaMMoreira) [reply packages/core/stories/state-icon.doc.stories.mdx:31]: Do you mean only on the first icon or all of them?

Comment (origami-z) [reply packages/core/stories/state-icon.doc.stories.mdx:31]: Could be all of the examples. Looks very tiny as a documentation example. @bhoppers2008 to comment :)

Maybe there is ways to set default props in mdx..?

File: packages/core/stories/state-icon.doc.stories.mdx (hunk 3 above)
Comment (JoanaMMoreira) [reply packages/core/stories/state-icon.doc.stories.mdx:48]: Yeah it's very odd, I'm looking into this

Comment (origami-z) [reply packages/core/stories/state-icon.doc.stories.mdx:48]: I may not be possible BTW, doc gen may not be able to trace through at all. If so, may be set the default value in state icon as well.

Comment (JoanaMMoreira) [reply packages/core/stories/state-icon.doc.stories.mdx:48]: It doesn't seem to be possible unfortunately, I've tried a few different ways but this seems to be a known issue in Storybook. I've exported the default `size` from `Icon` and added it to `StateIcon`

File: packages/core/stories/state-icon.doc.stories.mdx (hunk 2 above)
Comment (JoanaMMoreira) [reply packages/core/stories/state-icon.doc.stories.mdx:19]: Setting default props in mdx doesn't seem to work but I can add them individually to each example. @bhoppers2008 let me know your thoughts :)

Comment (bhoppers2008) [reply packages/core/stories/state-icon.doc.stories.mdx:19]: We can increase the size for documentation but would be best to align it to the work that paul is doing to remove size from the characteristics. That will have some size options. It's WIP though and Paul is on annual leave this week.

Comment (JoanaMMoreira) [reply packages/core/stories/state-icon.doc.stories.mdx:19]: Makes sense, so should we merge this as is for now and revisit later?

**Review by origami-z — APPROVED** [review]

**Review by bhoppers2008 — APPROVED** [review]

**Pull request merged**